RFC/src/RFC-0300_DAN.md

```diff
@@ -1 +1,179 @@
+that the rules of the asset contracts are enforced.
+* VNs form a peer-to-peer communication network that together defines the Tari [Digital Asset Network] (DAN)
+* VNs register themselves on the base layer and commit collateral to prevent Sybil attacks.
+* Scalability is achieved by sacrificing decentralisation. Not *all* VNs manage *every* asset. Assets are managed by
+  subsets of the DAN, called VN committees. These committees reach consensus on DA state amongst themselves.
+* VNs earn fees for their efforts.
+* Digital asset contracts are not Turing complete, but are instantiated by [Asset Issuer]s (AIs) using Digital Asset templates that are defined
+  in the DAN protocol code.
+
+### Digital Assets
+
+* Digital asset contracts are *not* Turing complete, but are selected from a set of [DigitalAssetTemplate]s that govern
+  the behaviour of each contract type. e.g. there could be a Single-Use Token template for simple ticketing systems; a
+  Coupon template for loyalty programmes and so on.
+* The template system is intended to be highly flexible and additional templates can be added to the protocol periodically.
+* Asset issuers can register TLDIs (top-level digital issuer) names on the base chain to help disambiguate similar
+  contracts and improve the signal-to-noise ratio from scam- or copy-cat contracts.
+
+An [Asset Issuer] (AI) will issue a Digital Assets by constructing a contract from on of the supported set of [DigitalAssetTemplate]s. The AI will choose
+ how large the committee of Validator Nodes will be for this DA and have the option to nominate [Trusted Node]s to be part of the VN committee for the DA.
+Any remaining spots on the committee will be filled by permissionless VNs that are selected randomly by the DAN.
+
+### Validator Nodes
+
+[Validator Node]s form the basis of the second layer DAN. All actions on this network take place by interacting with VN's. Some examples of actions
+that VNs will facilitate are:
+* Issuing a [Digital Asset],
+* querying the state of [Digital Asset] and its constituent [tokens],
+* issuing and instruction to change the state of a [Digital Asset] or [tokens].
+
+VNs will also perform archival function for the assets they manage. The lifetime of these archives and the fee structure for this function is
+still being discussed.
+
+#### Registration
+VNs register themselves on the [Base Layer] using a special [transaction] type. The registration [transaction] type
+requires spending of a certain minimum amount of [Tari coin], the ([RegistrationCollateral]), that has a time-lock on the
+output for a minimum amount of time ([RegistrationPeriod]) as well as some metadata, such as the VNs public key and a generated Node ID. The
+details of this transaction are not confidential and are publicly auditable. The Node ID is generated during registration to prevent mining
+of VN public keys that can be used to manipulate routing on the DAN.
```

Comment:
Some further suggestions on the details of this transaction:
* The blinding factor for the Registration transaction is the private key that the VN node will use to sign every instruction that it executes for the duration of its `RegistrationTerm` (or period)
* I've realised that the transaction *can* in fact be confidential, if the range_proof is modified to prove that the registration fee is in the range `[RegistrationCollateral; 2^64]`

Reply:
Added

Reply:
Removed the part about confidentiality, I think that might take some more thought because I think some parts of the transaction will definitely need to be auditable by the second layer?

Reply:
Agreed. Let's punt this.